api/models.py

```diff
@@ -87,7 +101,11 @@ def save(self, *args, **kwargs):
         if not self.item_code and not self.serial_number:
             self.full_clean()
         else:
-            super(Item, self).save(*args, **kwargs)
+            if (self.status in status for status in status_list):
+                super(Item, self).save(*args, **kwargs)
+
+            else:
+                raise ValueError('Status provided does not exist')
```

Comment:
thinking we should raise this one in `clean` as well, and do `full_clean` by default.

Reply:
I am not sure what you mean here...

Reply:
Do you mean we can do `raise ValueError('Status provided does not exist')` in the clean and then after all the if-else add `self.full_clean` in the `clean` ?

Reply:
@egichuri Do you mean like what I have pushed ?

Reply:
uhmm.. not exactly..
in `clean`:
```
if (self.status not in status for status in status_list):
            raise ValueError('Status provided does not exist')
```
and in `save`:
```
self.full_clean()
        super(Item, self).save(*args, **kwargs)
```

Reply:
also, note `status_list` is a list already, so `self.status not in status_list` should work